### DISCUSSION TITLE
Low resolution on live video stream

### DISCUSSION THREAD

**Discussion by jacoblannerbro** [blakeblackshear/frigate#3078] [Ask A Question]
Hi! Hope somebody could help with this. 
I'm getting low resolution on my live video stream and I can't figure out why.. The camera is capable of 2 k streaming. 
Here is my configuration file:

mqtt:
  host: 192.168.1.30
  user: *
  password: *
  stats_interval: 60

detectors: 
  cpu1:
    type: cpu
    num_threads: 2
  cpu2:
    type: cpu
    num_threads: 2
  cpu3:
    type: cpu
    num_threads: 2
  cpu4:
    type: cpu
    num_threads: 2
  
    
database:
  path: /media/frigate/frigate.db

cameras:
  forradet:
    ffmpeg:
      hwaccel_args:
        - -hwaccel
        - qsv
        - -qsv_device
        - /dev/dri/renderD128
      inputs:
        - path: rtsp://admin:*@192.168.3.109:554/Streaming/Channels/102/
          roles: 
            - detect
        - path: rtsp://admin:*@192.168.3.109:554/Streaming/Channels/101/
          roles:
            - record
            - rtmp
    detect:
      width: 640
      height: 360
      fps: 4
    objects:
      track:
        - person
        - car
        - dog
        - cat
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 7
    record:
      enabled: true
      events:
        retain: 
          default: 7

**Comment by NickM-27** [comment]
You are manually setting your resolution in detect, it should be set to the resolution of the camera

**Reply by jacoblannerbro** [reply]
But I thought that resolution was just for detection, it's on my substream with low resolution to keep CPU overhead low. My mainstream is full resolution and I want that to be shown when I choose to watch the live stream.

**Reply by NickM-27** [reply]
When you say "live stream" are you talking about the Frigate live view or streaming in HA / elsewhere?

**Reply by jacoblannerbro** [reply]
I mean when I click on the camera in Frigate and get the live feed.

**Reply by NickM-27** [reply]
Yeah, that uses the detect stream. This is because it takes a lot of CPU resources to decode the stream so if it did both streams while watching that would hit the CPU quite hard.

**Reply by jacoblannerbro** [reply]
Ok thanks for the clarification! 😀 But with this config it will still record in full resolution?

**Reply by NickM-27** [reply]
Yes, recording and the rtmp stream will use the high resolution stream

**Reply by jacoblannerbro** [reply]
Great! 👍😀

**Reply by Deishelon** [reply]
> Yeah, that uses the detect stream. This is because it takes a lot of CPU resources to decode the stream so if it did both streams while watching that would hit the CPU quite hard.

Is it something that can be overridden in the config?
I have cameras which output high and low res streams. I use high stream for recording and low for detect.
It will be great to have better live previews of the stream, and I'm ok with taking a cpu hit.

**Reply by NickM-27** [reply]
This is outdated, live view can be whatever stream you want

**Reply by Deishelon** [reply]
For anyone that searches how to do: https://docs.frigate.video/configuration/live#setting-stream-for-live-ui
